RealmSwift/Object.swift

```diff
@@ -410,13 +410,43 @@ public class ObjectUtil: NSObject {
         return nil
     }
 
-    // Get the names of all properties in the object which are of type List<>.
-    @objc private class func getGenericListPropertyNames(_ object: Any) -> NSArray {
-        return Mirror(reflecting: object).children.filter { (prop: Mirror.Child) in
-            return type(of: prop.value) is RLMListBase.Type
-        }.flatMap { (prop: Mirror.Child) in
-            return prop.label
-        } as NSArray
+    // Reflect an object, and remove all the ignored stored properties from the
+    // children collection.
+    private static func getNonIgnoredMirrorChildren(for object: Any) -> [Mirror.Child] {
+        guard let realmObject = object as? Object else {
+            return Array(Mirror(reflecting: object).children)
+        }
+        let ignoredPropNames: Set<String> = Set(type(of: realmObject).ignoredProperties())
+        // No HKT in Swift, unfortunately
+        return Mirror(reflecting: realmObject).children.filter { (prop: Mirror.Child) -> Bool in
```

Comment:
This probably also needs to handle lazy properties in some way (they produce a stored property that we have to explicitly filter out in other places).